Samples/Swift/nhubsample/nhubsample/ViewController.swift

```diff
@@ -0,0 +1,41 @@
+//
+//  Copyright © 2018 Microsoft All rights reserved.
+//  Licensed under the Apache License (2.0).
+//
+
+import UIKit
+
+class ViewController: UIViewController {
+    @IBOutlet weak var tagsTextField: UITextField!
+    @IBOutlet weak var registerButton: UIButton!
+    @IBOutlet weak var unregisterButton: UIButton!
+    
+    override func touchesBegan(_ touches: Set<UITouch>, with event: UIEvent?) {
+        // Simple method to dismiss keyboard when user taps outside of the UITextField.
+        self.view.endEditing(true)
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        // Do any additional setup after loading the view, typically from a nib.
+        
+        self.tagsTextField.text = UserDefaults.standard.string(forKey: Constants.NHUserDefaultTags)
+    }
+
+    @IBAction func handleRegister(_ sender: Any) {
+        // Save raw tags text in storage
+        UserDefaults.standard.set(self.tagsTextField.text, forKey: Constants.NHUserDefaultTags)
+        
+        //
+        // Delegate processing the register action to the app delegate.
+        //
+        let appDelegate: AppDelegate? = UIApplication.shared.delegate as? AppDelegate
```

Comment:
This should be moved to an extension of the ViewController and not in the AppDelegate. This is more idiomatic of swift as well as makes it easier to see how this is supposed to work.

Reply:
Done